### PR TITLE
feat: interactive mode in new cmd

### PR DESCRIPTION
**Pull request by ishanarya0**
Users can enable interactive mode by setting the `interactive` (shorthand `i`) flag.

### CODE REVIEW

**Comment by ravisuhag** [comment]
> The current command allows the user to pass blank `source` and `sink` as well, apart from the `processor`. Do we want to allow this? @ravisuhag @StewartJingga

I think, yes, Atleast one source and sink should be selected. @StewartJingga WDYT?

**Comment by StewartJingga** [comment]
> > The current command allows the user to pass blank `source` and `sink` as well, apart from the `processor`. Do we want to allow this? @ravisuhag @StewartJingga
> 
> I think, yes, Atleast one source and sink should be selected. @StewartJingga WDYT?

Yeah a source and a sink at least should be defined. For empty sink I wonder if we can use console as default? will that improve UX? @ravisuhag @ishanarya0

**Comment by StewartJingga** [comment]
Yeah a source and a sink at least should be defined. For empty sink I wonder if we can use console as default? will that improve UX? @ravisuhag @ishanarya0

Not implicitly when running of course, but when generating recipe. So if sink are not mentioned, we set console as the placeholder.

File: cmd/new.go [hunk 1]
```diff
@@ -95,6 +95,7 @@ func NewRecipeCmd() *cobra.Command {
 	}
 
 	cmd.Flags().StringVarP(&extractor, "extractor", "e", "", "Type of extractor")
+	cmd.MarkFlagRequired("extractor")
```
Comment (ravisuhag) [inline cmd/new.go:103]: @ishanarya0 Let's disable this rule.

**Comment by ravisuhag** [comment]
@ishanarya0 Instead of users enabling interactive mode. we should ask for the survey if they don't fill in information through flags.

File: cmd/new.go [hunk 2]
```diff
@@ -42,7 +45,8 @@ func NewRecipeCmd() *cobra.Command {
 
 			The recipe will be printed on standard output.
 			Specify recipe name with the first argument without extension.
-			Use commma to separate multiple sinks and processors.`),
+			Use commma to separate multiple sinks and processors.
+			Interactive mode can be enabled by setting --interactive or -i to true`),
```
Comment (ravisuhag) [inline cmd/new.go:49]: @ishanarya0 we can remove this line now.

**Pull request merged**